opendevin/runtime/utils/bash.py

```diff
@@ -1,49 +1,168 @@
-import bashlex
+def split_bash_commands(commands):
```

Comment:
why did you decide to revert the bash parsing in that PR? is it causing issues?

Reply:
> replaced split_bash_commands with a more versatile method to fix several issues

it is unclear as to what particular issue this will fix.. I'd actually prefer using a simpler solution (e.g., bashlex parser) rather than writing this complex stuff ourselves 🤣. What i can see is that after the modification, some of our parsing tests did not pass?

Reply:
If I could find it right now.... the version in main does give out ERROR messages on occasion, which imo shouldn't happen, right?

Reply:
Example:
https://github.com/OpenDevin/OpenDevin/actions/runs/10133503780/job/28026707072?pr=3158#step:8:1515
```py
02:37:00 - ACTION
**CmdRunAction (source=None)**
COMMAND:
ls -l
02:37:00 - OBSERVATION
**CmdOutputObservation (source=None, exit code=0)**
total 0
02:37:00 - ACTION
**IPythonRunCellAction**
CODE:
print('Hello, `World`!\n')
02:37:00 - opendevin:ERROR: bash.py:10 - Failed to parse bash commands
[input]: cat > /tmp/opendevin_jupyter_temp.py <<'EOL'
print('Hello, `World`!\n')
EOL
Error: : here-document at line 0 delimited by end-of-file (wanted "'EOL'") (position 76)
```

Reply:
This is not actually an error, rather than just a warning - it will fall back to the original multi-line command, which won't break all our evals.
I've corrected this in #3157 (change it to a debugging message)

Reply:
reverted